packages/node/src/lib/http-client.ts

```diff
@@ -0,0 +1,95 @@
+import { abortSignalAfterTimeout } from './abort'
+import { fetch as defaultFetch } from './fetch'
+
+/**
+ * This interface is meant to be compatible with different fetch implementations (node and browser).
+ * Using the ambient fetch type is not possible because the AbortSignal type is not compatible with node-fetch.
+ * @link https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API
+ */
+export interface HTTPFetchFn {
+  (url: string, requestInit: HTTPFetchRequest): Promise<HTTPResponse>
+}
+
+/**
+ * This interface is meant to be compatible with the Request interface.
+ * @link https://developer.mozilla.org/en-US/docs/Web/API/Request
+ */
+export interface HTTPFetchRequest {
+  headers?: Record<string, string>
+  body?: string
+  method?: string
+  signal?: any // AbortSignal type does not play nicely with node-fetch
+}
+
+/**
+ * This interface is meant to very minimally conform to the Response interface.
+ * @link https://developer.mozilla.org/en-US/docs/Web/API/Response
+ */
+export interface HTTPResponse {
+  ok: boolean
```

Comment:
Kind of wonder if we really need `ok` - it's just a boolean that is `true` if `status` if 200-299. We get `status` already so could check that ourselves (then if someone uses an HTTP client that doesn't return `ok`, they don't have to worry about it)